src/rp2_common/hardware_dma/dma.c

```diff
@@ -70,6 +70,20 @@ bool dma_timer_is_claimed(uint timer) {
     return hw_is_claimed(&_timer_claimed, timer);
 }
 
+void dma_channel_cleanup(uint channel) {
+    check_dma_channel_param(channel);
+    // we definitely want to disable chaining, because that could cause a restart
+    // during abort. Sicne we're doing that, it's just as easy to reset to default config!
+    const dma_channel_config config = dma_channel_get_default_config(channel);
```

Comment:
What state is the channel expected to be in when this is called? Wiping the config when it is potentially running seems unsafe.

Possibly this should be:

```c
// Disable CHAIN_TO, and disable channel, so that it ignores any further triggers
hw_write_field(
    &dma_hw->ch[channel].al1_ctrl,
    (channel << DMA_CH0_CTRL_TRIG_CHAIN_TO_LSB) | (0u << DMA_CH0_CTRL_TRIG_EN_LSB),
    DMA_CH0_CTRL_TRIG_CHAIN_TO_BITS | DMA_CH0_CTRL_TRIG_EN_BITS
);
```

What do you think?

Reply:
We call this when we've finished using it when uninitialising the wifi driver that no longer needs the dma channel or pios

Reply:
If it's always expected to be called after the channel has been brought to a halt, then the current code is fine, but the abort seems unnecessary, and it's probably worth making this explicit in the doc comment in dma.h.

If it's expected to be possible to call this function at any time, then it's probably not a good idea to write all fields in the CTRL register. For example, this code will turn on read increment, which means there is a period where the channel may read from unexpected addresses (like other PIO FIFOs) before we call `dma_channel_abort()` after clearing the IRQs.

Reply:
I see, that sounds bad. Your change makes sense, although I assume you meant hw_write_masked.

Reply:
> although I assume you meant hw_write_masked
yes, oops